src/parser/jobs/nin/modules/TrickAttackUsage.tsx

```diff
@@ -0,0 +1,126 @@
+import {Trans, Plural} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Downtime from 'parser/core/modules/Downtime'
+import Suggestions, {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const OPTIMAL_GCD_COUNT = 5 // Opener should be Suiton > AE combo > SE before Trick
+
+const MUDRAS: ActionKey[] = [
+	'TEN',
+	'TEN_KASSATSU',
+	'CHI',
+	'CHI_KASSATSU',
+	'JIN',
+	'JIN_KASSATSU',
+]
+
+export class TrickAttackUsage extends Analyser {
+	static override handle = 'taUsage'
+
+	@dependency private data!: Data
+	@dependency private downtime!: Downtime
+	@dependency private suggestions!: Suggestions
+
+	private mudraActions: number[] = []
+
+	private taCasts: number[] = []
+	private lostTime: number = 0
+	private gcdCount: number = 0
+	private castHook?: EventHook<Events['action']>
+
+	override initialise() {
+		this.mudraActions = MUDRAS.map(k => this.data.actions[k].id)
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.castHook = this.addEventHook(playerFilter.type('action'), this.onCast)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.TRICK_ATTACK.id), this.onTrickAttack)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']) {
+		const action = this.data.getAction(event.action)
+		if (event.timestamp >= this.parser.fight.start_time && action && action.onGcd && this.mudraActions.indexOf(action.id) === -1) {
+			// Don't count the individual mudras as GCDs for this - they'll make the count screw if Suiton wasn't set up pre-pull
+			this.gcdCount++
+		}
+	}
+
+	private onTrickAttack(event: Events['action']) {
+		if (this.castHook != null) {
+			this.removeEventHook(this.castHook)
+			this.castHook = undefined
+		}
+
+		if (this.taCasts.length > 0) {
+			const lastCast = this.taCasts[this.taCasts.length - 1]
+			const taAvailable = lastCast + this.data.actions.TRICK_ATTACK.cooldown
+			const downtime = this.downtime.getDowntime(
+				this.parser.fflogsToEpoch(taAvailable),
+				this.parser.fflogsToEpoch(event.timestamp),
+			)
+			this.lostTime += Math.max((event.timestamp - taAvailable) - downtime, 0)
```

Comment:
With a history of windows you could shift this to the onComplete callback and do it all at once, I'd expect that's faster. Noted the `indexOf` rather than `includes` above so I figure speed is of some concern here.

Reply:
I can look at moving this around. I'm not sure it'll be meaningfully faster but it might be nice for logical separation to have the math happen at the end and have the event hooks just worry about the specific event in question.

Reply:
Sounds good to me.